From fe77f4794025deb7811e65f6f77d79b1be508fe5 Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Mon, 16 Feb 2015 17:11:31 +0100
Subject: [PATCH 03/16] Revert "COMPATIBILITY BREAK: remove --uuid (use
 standard -n / --plug instead)"

This reverts commit 1204c484a089ba337c6d4c9797eafce24a71915a.

Conflicts:
	fence/agents/lib/fencing.py.py
	fence/agents/vmware_soap/fence_vmware_soap.py
	fence/agents/xenapi/fence_xenapi.py
---
 fence/agents/lib/fencing.py.py                | 20 ++++++++++++++++++++
 fence/agents/vmware_soap/fence_vmware_soap.py |  2 +-
 fence/agents/xenapi/fence_xenapi.py           |  2 +-
 3 files changed, 22 insertions(+), 2 deletions(-)

diff --git a/fence/agents/lib/fencing.py.py b/fence/agents/lib/fencing.py.py
index 994a562..d4bb866 100644
--- a/fence/agents/lib/fencing.py.py
+++ b/fence/agents/lib/fencing.py.py
@@ -371,6 +371,13 @@ all_opt = {
 		"required" : "1",
 		"shortdesc" : "The URL of the XenServer host.",
 		"order" : 1},
+	"uuid" : {
+		"getopt" : "U:",
+		"longopt" : "uuid",
+		"help" : "-U, --uuid                     UUID of the VM to fence",
+		"required" : "0",
+		"shortdesc" : "The UUID of the virtual machine to fence.",
+		"order" : 1},
 	"sudo" : {
 		"getopt" : "",
 		"longopt" : "use-sudo",
@@ -631,8 +638,21 @@ def check_input(device_opt, opt):
 	if options["--action"] == "disable":
 		options["--action"] = "off"
 
+	# UUID is now only alias for --plug; because we can detect it automatically
+	if options.has_key("--uuid"):
+		options["--plug"] == options["--uuid"]
+		del options["--uuid"]
+
 	_validate_input(options)
 
+	## automatic detection and set of valid UUID from --plug
+	try:
+		options["--uuid"] = str(uuid.UUID(options["--plug"]))
+	except ValueError:
+		pass
+	except KeyError:
+		pass
+
 	if options.has_key("--debug-file"):
 		try:
 			debug_file = logging.FileHandler(options["--debug-file"])
diff --git a/fence/agents/vmware_soap/fence_vmware_soap.py b/fence/agents/vmware_soap/fence_vmware_soap.py
index d1f6c00..5f760cc 100644
--- a/fence/agents/vmware_soap/fence_vmware_soap.py
+++ b/fence/agents/vmware_soap/fence_vmware_soap.py
@@ -215,7 +215,7 @@ def logout():
 def main():
 	global options_global
 	global conn_global
-	device_opt = ["ipaddr", "login", "passwd", "web", "ssl", "notls", "port"]
+	device_opt = ["ipaddr", "login", "passwd", "web", "ssl", "notls", "port", "uuid"]
 
 	atexit.register(atexit_handler)
 	atexit.register(logout)
diff --git a/fence/agents/xenapi/fence_xenapi.py b/fence/agents/xenapi/fence_xenapi.py
index 5fe7646..1afb1df 100644
--- a/fence/agents/xenapi/fence_xenapi.py
+++ b/fence/agents/xenapi/fence_xenapi.py
@@ -199,7 +199,7 @@ def return_vm_reference(session, options):
 
 def main():
 
-	device_opt = ["login", "passwd", "port", "no_login", "no_password", "session_url", "web"]
+	device_opt = ["login", "passwd", "port", "no_login", "no_password", "session_url", "web", "uuid"]
 
 	atexit.register(atexit_handler)
 
-- 
1.9.3

